stock_custom_exchange_rate_date/models/stock_move.py

```diff
@@ -0,0 +1,28 @@
+from odoo import api, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    @api.multi
+    def _get_price_unit(self):
+        """Take into account custom rate date, if provided"""
+        line = self.purchase_line_id
+        if (not self.picking_id.exchange_rate_date
+                or not line
+                or self.product_id != line.product_id
+                or line.currency_id == line.company_id.currency_id):
+            return super()._get_price_unit()
```

Comment:
try 119 length of line please :-(

Reply:
You mean the if or the super?

Reply:
Done. I personally liked a little bit more the previous form, though ;).